node/src/components/consensus/highway_core/highway.rs

```diff
@@ -512,6 +507,9 @@ impl<C: Context> Highway<C> {
                         .validators
                         .id(*creator)
                         .ok_or(EndorsementError::Creator)?;
+                    if self.state.opt_fault(*creator) == Some(&Fault::Banned) {
+                        return Err(EndorsementError::Banned.into());
```

Comment:
What if the creator of the `Vertex::Endorsements` didn't see the creator as faulty like we do?

Reply:
We already filter out endorsements from faulty validators in `add_endorsements`.

Reply:
`Banned` means that the previous switch block already saw them as faulty: Every validator will have the same `Banned` entries and ignore everything that comes from a banned validator.